File: 006/006.opt.py
def sqdiff(limit):
  ''' Calculates the difference between sum of squares
  and square of sum for all natural numbers in closed interval. \n
    Args:
      limit: the end of the interval, [1, limit]
    Returns:
      An integer.  '''

  sumsq = lambda limit: (2 * limit**3 + 3 * limit**2 + limit) / 6  # sum of squres  
  sqsum = lambda limit: ((1 + limit) * limit / 2) ** 2             # squre of sum
  return int(sqsum(limit) - sumsq(limit))

# tests
# print(sqdiff(100))

File: deprecated/006/js/006.opt.js
function squareDifference(limit) {
  /**
   * Calculates the difference for all natural numbers between
   * squared sum and sum of squares
   * @param {Integer} limit The last natural number to calculates sums for
   * @return {Integer} The calculated difference
   */
  let squareSum  = x => (1 + limit) * limit / 2;
  let sumSquares = x => (2 * x**3 + 3 * x**2 + x) / 6;
  return squareSum(limit) ** 2 - sumSquares(limit);
}

console.log(squareDifference(100));